### PR TITLE
Fix bugs in cutout training

### PR DESCRIPTION
**Pull request by ravinkohli**
This PR fixes the following bugs:

1. we were not sampling the indices with replacement.
2. we were taking the minimum between batch size and num features, which is not necessary
3. Numerical features were getting -1 to them, whereas 0 makes more sense

### CODE REVIEW

File: autoPyTorch/pipeline/components/training/trainer/RowCutMixTrainer.py [hunk 1]
```diff
@@ -39,7 +39,8 @@ def data_preparation(self, X: np.ndarray, y: np.ndarray,
         # It is unlikely that the batch size is lower than the number of features, but
         # be safe
         size = min(X.shape[0], X.shape[1])
```
Comment (ArlindKadra) [inline autoPyTorch/pipeline/components/training/trainer/RowCutMixTrainer.py:41]: This should also be changed to size=X.shape[1] right ?

Comment (ravinkohli) [reply autoPyTorch/pipeline/components/training/trainer/RowCutMixTrainer.py:41]: True, I missed that

File: autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py [hunk 2]
```diff
@@ -34,17 +36,26 @@ def data_preparation(self, X: np.ndarray, y: np.ndarray,
...
-        # It is unlikely that the batch size is lower than the number of features, but
-        # be safe
-        size = min(X.shape[0], X.shape[1])
-        indices = torch.tensor(self.random_state.choice(range(1, size), max(1, np.int(size * self.patch_ratio))))
+        size = X.shape[1]
+        indices = self.random_state.choice(range(1, size), max(1, np.int32(size * self.patch_ratio)),
+                                           replace=False)
 
-        # We use an ordinal encoder on the tabular data
+        if not isinstance(self.numerical_columns, typing.Iterable):
```
Comment (ArlindKadra) [inline autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py:43]: What if the numerical columns are None, we should still continue with only categorical imputing in this case or not.

Comment (ArlindKadra) [reply autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py:43]: Also if there are only numerical columns, there should not be a conversion for categorical ones.

Comment (ravinkohli) [reply autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py:43]: Actually when there are no numerical columns, it is not none but it is an empty list. And indexing with an empty list does not affect the tensor so this should work

Comment (ArlindKadra) [reply autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py:43]: https://github.com/automl/Auto-PyTorch/blob/8b71ee2ad3c00277928ac462c0d364151d69aca0/autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py#L340-L341

Is 'numerical_columns' always in 'dataset_properties' ?

Comment (ravinkohli) [reply autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py:43]: when its tabular data then yeah

**Pull request merged**